pkg/operator/storageclasscontroller/vmware.go

```diff
@@ -135,12 +155,18 @@ func (v *storagePolicyAPI) createZonalStoragePolicy(ctx context.Context) (string
 
 	var aggregatedErrors []error
 
+	// Failure domains go across vCenters.  So really we want to only do failure domains that are for this connection's
+	// vCenter.
 	for _, failureDomain := range failureDomains {
-		dataCenter := failureDomain.Topology.Datacenter
-		dataStore := failureDomain.Topology.Datastore
-		err := v.attachTags(ctx, dataCenter, dataStore)
-		if err != nil {
-			aggregatedErrors = append(aggregatedErrors, err)
+		klog.V(4).Infof("Is failure domain %v part of this connection (%v): %v", failureDomain.Name, v.vcenterApiConnection.Hostname, failureDomain.Server == v.vcenterApiConnection.Hostname)
```

Comment:
Rather than doing this check here, we should ensure that we never run into a scenario that - this code is suuplied incorrect connection object.

Reply:
Ok.  I can enhance the connection ctx to contain the FDs that are related to that connection.  I can populate this around connection creation time after we log into the connections.  Would that be a good approach?

Reply:
I was thinking that - may be, we should first find all failure-domains which are on this vcenter and then iterate through them. That way, the intent of the code will be clearer to the reader that, we are trying to create storage policy on failure-domains associated with this vcenter.

Reply:
Excellent.  I like that.  I'll mock that up.

Reply:
Ok, I have made changes for this.  Please feel free to review and let me know your thoughts.